examples/sampling/node_classification.py

```diff
@@ -0,0 +1,284 @@
+large graphs using efficient neighbor sampling.
+
+Paper: [Inductive Representation Learning on Large Graphs]
+(https://arxiv.org/abs/1706.02216)
+
+If you want a deeper understanding of node classification. You can
+read the example in the `examples/pytorch/graphsage/node_classification.py`
+TODO(#5797): Move `graphsage/node_classification.py` to the `examples/core/`.
+
+This flowchart describes the main functional sequence of the provided example.
+main
+│
+├───> Load and preprocess dataset
+│
+├───> Instantiate SAGE model
+│
+├───> train
+│     │
+│     ├───> NeighborSampler (HIGHLIGHT)
+│     │
+│     └───> Training loop
+│           │
+│           └───> SAGE.forward
+│
+└───> layerwise_infer
+      │
+      └───> SAGE.inference
+            │
+            └───> MultiLayerFullNeighborSampler (HIGHLIGHT)
+"""
+
+import argparse
+
+import dgl
+import dgl.nn as dglnn
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchmetrics.functional as MF
+import tqdm
+from dgl.data import AsNodePredDataset
+from dgl.dataloading import (
+    DataLoader,
+    MultiLayerFullNeighborSampler,
+    NeighborSampler,
+)
+from ogb.nodeproppred import DglNodePropPredDataset
+
+
+class SAGE(nn.Module):
+    def __init__(self, in_size, hidden_size, out_size):
+        super().__init__()
+        self.layers = nn.ModuleList()
+        # Three-layer GraphSAGE-mean.
+        self.layers.append(dglnn.SAGEConv(in_size, hidden_size, "mean"))
+        self.layers.append(dglnn.SAGEConv(hidden_size, hidden_size, "mean"))
+        self.layers.append(dglnn.SAGEConv(hidden_size, out_size, "mean"))
+        self.dropout = nn.Dropout(0.5)
+        self.hidden_size = hidden_size
+        self.out_size = out_size
+
+    def forward(self, blocks, x):
+        hidden_x = x
+        for layer_idx, (layer, block) in enumerate(zip(self.layers, blocks)):
+            hidden_x = layer(block, hidden_x)
+            is_last_layer = layer_idx == len(self.layers) - 1
+            if not is_last_layer:
+                hidden_x = F.relu(hidden_x)
+                hidden_x = self.dropout(hidden_x)
+        return hidden_x
+
+    def inference(self, g, device, batch_size):
+        """Conduct layer-wise inference to get all the node embeddings."""
+        feat = g.ndata["feat"]
+        #####################################################################
+        # (HIGHLIGHT) Creating a MultiLayerFullNeighborSampler instance.
+        # This sampler is used in the Graph Neural Networks (GNN) training
+        # process to provide neighbor sampling, which is crucial for
+        # efficient training of GNN on large graphs.
+        #
+        # The first argument '1' indicates the number of layers for
+        # the neighbor sampling. In this case, it's set to 1, meaning
+        # only the direct neighbors of each node will be included in the
+        # sampling.
+        #
+        # The 'prefetch_node_feats' parameter specifies the node features
```

Comment:
With these comment, it is still not very clear how prefetch_node_feats happens, even with this doc (https://docs.dgl.ai/en/latest/guide/minibatch-prefetching.html).

@BarclayII can you chime in here?

Reply:
In the following text, the `prefetch` parameter is also used, and I've added some advantages of using `prefetch` there. I've done some research and found this blog post to be quite detailed: https://discuss.dgl.ai/t/confusion-about-feature-prefetch/3573/4. In summary, it appears to be a technique used to speed up I/O operations.

Reply:
https://docs.google.com/document/d/1uyUfRc9SuavR-_Nr-Rn4acEehD6cUE5oTjXHPmUkfWI/edit#bookmark=id.yp7uomdejg6w

Here is the explanation of prefetch, can you update the documentation accordingly?